src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -902,6 +953,61 @@ private ImmutableHashSet<ISymbol> ComputeGeneratedCodeSymbolsInTree(SyntaxTree t
             return generatedSymbolsBuilderOpt != null ? generatedSymbolsBuilderOpt.ToImmutable() : ImmutableHashSet<ISymbol>.Empty;
         }
 
+        private bool IsAnalyzerSuppressedForTree(DiagnosticAnalyzer analyzer, SyntaxTree tree)
+            => GetOrComputeSuppressedAnalyzersForTree(tree).Contains(analyzer);
```

Comment:
I don't understand what's going on here. It looks like we build a hash set, then do a single `Contains` call, then throw the hash set away. Is that efficient?

Reply:
Hash set is saved into `_lazySuppressedAnalyzersForTreeMap`. I will change this method to `return _lazySuppressedAnalyzersForTreeMap.GetOrAdd(tree, computeSuppressedAnalyzersForTree).Contains(analyzer);` and also move up the local function `computeSuppressedAnalyzersForTree` for improved readability.